Add a part about custom style in markdown for DOCX reference doc (#975)
quarto-dev/quarto-web: docs/output-formats/ms-word-templates.qmd

![](images/word-modify-styles.png){fig-alt="Modify Style dialog box with a properties section for selecting what is to be styled, and a formatting section for selecting the style settings. The color selection dropdown in the formatting section is open."}

After you finish modifying the styles, you can save the document and use it as the template for future Word documents.

## Applying Custom Styles

Pandoc provides default styles for elements like paragraphs or code blocks and default formatting for inline elements like bold. Custom styles can be defined using divs for blocks and spans for text, so a specific style is used instead of the default. 

Set the `custom-style` attribute with the name of the style defined in the default or user-provided reference document. 

> For example, this applies a text style called `Emphasis` to the `Get out` text instead of the default style for text.

Reviewer: ```suggestion
For example, this applies a text style called `Emphasis` to the `Get out` text instead of the default style for text:
```

```markdown
[Get out]{custom-style="Emphasis"}, he said.
```

> This will apply the paragraph style called `Poetry` to the content inside the Divs syntax `:::`

Reviewer: ```suggestion
This will apply the paragraph style called `Poetry` to the content inside the Div (`:::`):
```

```markdown
Dickinson starts the poem simply:

::: {custom-style="Poetry"}
| A Bird came down the Walk---
| He did not know I saw---
:::
```

> If the reference.docx file does not already contain the styles, they will be added to the output file with standard text as their base style. In case the styles are already defined, their definition will not be modified.

Reviewer: ```suggestion
If the `reference.docx` file does not already contain the styles, they will be added to the output file with standard text as their base style. If the styles are already defined, their definitions will not be modified.
```

> A possible workflow would be 

Reviewer: ```suggestion
A possible workflow could be:
```

> - Create a document using `custom-style` you need,
> - Render this document so that they are created,
> - Save this document as a reference document for further rendering

Reviewer: ```suggestion
1. Create a document setting `custom-style` for all the styles you need,
2. Render this document so that the styles are created in the output document, 
3. Save this document as a reference document for future rendering
```

> This ensures that the names of the styles used in `custom-style` match the names of the styles defined in the reference document.

Reviewer: ```suggestion
This ensures that the names of the styles used in `custom-style` exactly match the names of the styles defined in the reference document.
```

> Overall, this mechanism allows for great customization as it also works with a new style that would be defined in the reference doc. For advanced users, this could be coupled with [filters](../extensions/filters.qmd) to apply `custom-style` attribute to a filtered set of element in the document. 

Reviewer: ```suggestion
Overall, this mechanism allows for great customization by allowing authors to define new styles in the reference document. For advanced users, this could be coupled with [filters](../extensions/filters.qmd) to apply a `custom-style` attribute to a filtered set of elements in the document. 
```
